internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/restart/TestRestartServerAfterDeletingSstFilesWithActiveActiveIngestion.java

```diff
@@ -439,50 +511,97 @@ public void testActiveActiveStoreWithRMDAndRestartServer(boolean deleteSSTFiles,
       // check setHybridRewindSeconds() config in setup
       for (int key: allIncPushKeys) {
-        AvroGenericStoreClient<String, Object> finalStoreClient2 = storeClient;
         TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, () -> {
-          assertNotNull(finalStoreClient2.get(KEY_PREFIX + key).get());
-          assertEquals(finalStoreClient2.get(KEY_PREFIX + key).get().toString(), VALUE_PREFIX_INC_PUSH + key);
+          assertNotNull(finalStoreClient.get(KEY_PREFIX + key).get());
+          assertEquals(finalStoreClient.get(KEY_PREFIX + key).get().toString(), VALUE_PREFIX_INC_PUSH + key);
         });
       }
     } finally {
       if (storeClient != null) {
         storeClient.close();
       }
     }
+
+    if (!deleteSSTFiles && !deleteRMDSSTFiles) {
+      /** creating a {@link pubSubConsumer} to check for the messages with
+       * {@link VENICE_LEADER_COMPLETION_STATUS_HEADER} header. Testing this for just 1 case
+       * should be enough as deleting sst and restarting servers doesn't affect the below generic
+       * test as we are consuming from kafka to see if the new header is found or not. More specific
+       * tests can be added for those cases if needed.
+       */
+      Properties properties = new Properties();
+      properties.setProperty(
+          ConfigKeys.KAFKA_BOOTSTRAP_SERVERS,
+          clusterWrappers.get(NON_SOURCE_COLO).getPubSubBrokerWrapper().getAddress());
+      pubSubConsumer = pubSubBrokerWrapper.getPubSubClientsFactory()
+          .getConsumerAdapterFactory()
+          .create(new VeniceProperties(properties), false, PubSubMessageDeserializer.getInstance(), "testConsumer");
+      pubSubConsumer
+          .subscribe(new PubSubTopicPartitionImpl(new PubSubTopicRepository().getTopic(topic), PARTITION_ID), 0);
+
+      AtomicBoolean isLeaderCompletionHeaderFound = new AtomicBoolean(false);
+      AtomicBoolean isLeaderCompleted = new AtomicBoolean(false);
+      TestUtils.waitForNonDeterministicAssertion(20, TimeUnit.SECONDS, () -> {
+        Map<PubSubTopicPartition, List<PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long>>> messages =
+            pubSubConsumer.poll(10 * Time.MS_PER_SECOND);
+        for (Map.Entry<PubSubTopicPartition, List<PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long>>> entry: messages
+            .entrySet()) {
+          List<PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long>> pubSubMessages = entry.getValue();
+          for (PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> message: pubSubMessages) {
+            PubSubMessageHeaders pubSubMessageHeaders = message.getPubSubMessageHeaders();
+            for (PubSubMessageHeader header: pubSubMessageHeaders.toList()) {
+              if (header.key().equals(VENICE_LEADER_COMPLETION_STATE_HEADER)) {
+                isLeaderCompletionHeaderFound.set(true);
+                if (LeaderCompleteState.valueOf(header.value()[0]) == LEADER_COMPLETED) {
```

Comment:
@gaojieliu added this check in an existing test to check whether the new header is being sent and validating the values. Will be adding more tests in the 2nd diff.